### DISCUSSION TITLE
What are props in React?

### DISCUSSION THREAD

**Discussion by Akash2377** [Akash2377/Akash2377_portfolio#13] [Q&A] [answered]
What are props in React?

**Comment by SaurabhSinghRbz** [comment, marked as answer]
In [ReactJS](https://www.simplilearn.com/tutorials/reactjs-tutorial/what-is-reactjs), the props are a type of object where the value of attributes of a tag is stored. The word “props” implies “properties”, and its working functionality is quite similar to [HTML](https://www.simplilearn.com/tutorials/html-tutorial/what-is-html) attributes. 

Basically, these props components are read-only components. In ReactJS, the data can be passed from one component to another component using these props, similar to how the arguments are passed in a function. Inside the component, we can add the attributes called props; however, we cannot change or modify props inside the component as they are immutable.

Using the “this.props”, we can make the props available inside the components. Then, the dynamic data can be rendered through the render method. We need to add the props to the reactDOM.render() in the Main.js file of our ReactJS project of ReactJS if we need immutable data in the component. Then we can use it in the component where we want to use those dynamic data.

Example: ->
The code below is the Main.js file. Here, we pass the dynamic data for name attribute within the App component through the render method called reactDOM.render. 

import React from 'react';  

import ReactDOM from 'react-dom';  

import App from './App.js';  

ReactDOM.render(<App name = "World of Programming!" />, document.getElementById('app'));  

The below code is for App.js. Here, we are using the dynamic data for the name attribute which the Main.js file has passed as props, and making the props available within this App component using “this.props”.

import React, { Component } from 'react';  

class App extends React.Component {  

   render() {    

      return (  

          <div>  

            <h1> Welcome to the { this.props.name } </h1>    

            <p> <h4> Here you will get to know so many things and your knowledge will be enhanced . </h4> </p>          

          </div>  

      );  

   }  

}  

export default App;  

Now let us discuss the default props in ReactJS.

**Comment by abhishekpandey108** [comment]
We use props in React to pass data from one component to another component.
